flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -289,20 +297,44 @@ def real_nodes(self):
     def real_edges(self):
         """Returns an iterable of all edges excluding the ones connected to the
         'low' or 'high' points."""
-        is_high_or_low = lambda n: n in ("low", "high")
         return filter(
-            lambda edge: not (is_high_or_low(edge[0]) or is_high_or_low(edge[1])),
+            lambda edge: edge[0] not in ("low", "high") and edge[1] not in ("low", "high"),
             self.edges(),
         )
 
+    def assign_weights(self, code):
+        """Assign the weights to the graph based on the weight
+        of the common vertex of each stabilizer pair of the code."""
+        for edge in self.edges():
+            data = self.edge_data(*edge)
+            if data["common_vertex"] is not None:
+                data["weight"] = code.graph.nodes[data["common_vertex"]].get("weight")
+            elif "high" in edge or "low" in edge:
```

Comment:
Would this be slightly better:
```suggestion
            elif edge[0] in ("low", "high"):
```

Reply:
What of edge[1] is High or low?

Reply:
I think that - in practice - (see e.g. add_low_bound_point, add_high_bound_point) these are always in edge[0]. This means the real_edges method can also be modified.

However, to be extra careful we can keep it as is. What do you think?

Reply:
I think we should be conservative and check both side of the edge.